src/canvas/Canvas.ts

```diff
@@ -1516,7 +1494,6 @@ export class Canvas extends SelectableCanvas {
         }
         this._fireSelectionEvents(prevActiveObjects, e);
       } else {
-        isInteractiveTextObject(activeObject) && activeObject.exitEditing();
```

Comment:
should be handled by

https://github.com/fabricjs/fabric.js/blob/8eb642460556afea542e50537e1bcf0de3cb79bb/src/canvas/SelectableCanvas.ts#L1435-L1438


https://github.com/fabricjs/fabric.js/blob/8eb642460556afea542e50537e1bcf0de3cb79bb/src/shapes/IText/ITextBehavior.ts#L109-L113

Reply:
And is more correct since if onDeselect returns true then text editing should remain active... and perhaps multi selection should abort.

Reply:
so to male things clear, is this the only thing you disapprove of?

Reply:
this line i would put back

Reply:
```ts
(activeObject as IText).exitEditing && activeObject.exitEditing();
```
it think something like this would just work fine and not introduce a change we don't want to think about now

Reply:
You want to put it back because you are afraid of a regression?
I understand your point I think.
But disagree.
Cleaning up has many gains. And leaving a dead/redundant line of code is a future bug in potential. And may cause developing other stuff seem harded because it will be viewed as a constrain when actually it should not be there.
I am willing the test it deeply.

Since you are unwilling then we should at least add a TODO comment explaining this. I am OK with that.

Reply:
> You want to put it back because you are afraid of a regression?

Yes removing a single line and adding a possible regression when we are trying to clean up is risky.
Right now in the worst case we are calling exit editing twice. Is not a big deal.
As we discussed we can keep this, and just make a comprehensive plan on who should handle the deselection side effect and how and let's think of this.